prov/verbs/src/verbs_info.c

```diff
@@ -791,21 +794,51 @@ static int vrb_alloc_info(struct ibv_context *ctx, struct fi_info **info,
 
 	switch (ctx->device->transport_type) {
 	case IBV_TRANSPORT_IB:
-		if (ibv_query_gid(ctx, 1, 0, &gid)) {
+		if (ibv_query_gid(ctx, 1, vrb_gl_data.gid_idx, &gid)) {
 			VERBS_INFO_ERRNO(FI_LOG_FABRIC,
 					 "ibv_query_gid", errno);
 			ret = -errno;
 			goto err;
 		}
 
-		name_len = strlen(VERBS_IB_PREFIX) + INET6_ADDRSTRLEN;
-		if (!(fi->fabric_attr->name = calloc(1, name_len + 1))) {
-			ret = -FI_ENOMEM;
+		if (ibv_query_gid_type(ctx, 1, vrb_gl_data.gid_idx, &gid_type)) {
+			VERBS_INFO_ERRNO(FI_LOG_FABRIC,
+					 "ibv_query_gid_type", errno);
+			ret = -errno;
 			goto err;
 		}
+		switch (gid_type) {
+		case IBV_GID_TYPE_IB_ROCE_V1:
+			name_len = strlen(VERBS_ROCEV1_PREFIX) + INET6_ADDRSTRLEN;
+			if (!(fi->fabric_attr->name = calloc(1, name_len + 1))) {
+				ret = -FI_ENOMEM;
+				goto err;
+			}
 
-		snprintf(fi->fabric_attr->name, name_len, VERBS_IB_PREFIX "%" PRIx64,
-			 be64toh(gid.global.subnet_prefix));
+			snprintf(fi->fabric_attr->name, name_len,
+				 VERBS_ROCEV1_PREFIX "%" PRIx64,
+				 be64toh(gid.global.subnet_prefix));
+			break;
+		case IBV_GID_TYPE_ROCE_V2:
+			name_len = strlen(VERBS_ROCEV2_PREFIX) + INET6_ADDRSTRLEN;
+			if (!(fi->fabric_attr->name = calloc(1, name_len + 1))) {
+				ret = -FI_ENOMEM;
+				goto err;
+			}
+
+			snprintf(fi->fabric_attr->name, name_len,
+				 VERBS_ROCEV2_PREFIX "%" PRIx64,
+				 be64toh(gid.global.subnet_prefix));
```

Comment:
Is there no meaningful value for RoCE GIDs that we can record?  The description shows that this just returns 0.

Reply:
This was actually the reason I posted the linked issue #6365. The idea was to have the IPv4 RoCE devices to list their respective IPv4 addresses after the prefixes. Converting the IPv6-formatted GID to IPv4 and having that print isn't a good choice, as more than 1 device can share a single GID index and the GIDs are currently being queried via GID index. The idea of reassigning the verbs_fabric_attr value to something specific to the fabric rather than to the provider has been floated -- instead of passing back the global info for the Verbs provider to return the specific GID, tailor the fabric_attr field to contain something fabric-specific and populate the information into that per-fabric data structure. Thoughts towards how to implement this? Or different ideas entirely?

Reply:
If we can use the IP address, I would be fine with that.  The tcp/udp providers do this, so there's code available to discover IP addresses and convert them.  I'm okay even removing the 'RoCEv2' portion of the string in that case, since ideally the protocol conveys that same information.

Do we know how the RoCEv2 GID value is formed?  Is the IP address buried in that, or does that contain some MAC address?

Reply:
IIRC, the GID value is formed from the IPv4 or IPv6 address associated with the gid. It's the reason why GID 0 is often 0xfe800000 (IPv6 autoconf address). For IPv6, the assumption to date has been they always have a 64 bit prefix. For IPv4, we didn't really encounter the gid parsing because of the default assumption of GID 0.

Reply:
If that's the case, then trying to use the IP address should basically result in the same value, right?  I.e. we would still get the IPv6 autoconf address.

Reply:
Yes, I think it should work.